cmd/thanos/compact.go

```diff
@@ -168,10 +168,17 @@ func runCompact(
 		Name: "thanos_compactor_retries_total",
 		Help: "Total number of retries after retriable compactor error",
 	})
+	iterations := prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "thanos_compactor_iterations_total",
+		Help: "Total number of iterations that were executed successfully",
+	})
 	halted.Set(0)
 
 	reg.MustRegister(halted)
 	reg.MustRegister(retried)
+	if wait {
```

Comment:
I think we can register all anyway - but not a big deal

Reply:
If it's not waiting it's a batch job so metrics will not be important anyway.

Reply:
Sure, removed this. I was being extra resource-conscious.